server/src/main/java/org/elasticsearch/search/rank/RankDoc.java

```diff
@@ -21,7 +21,7 @@
  * Subclasses should extend this with additional information
  * required for their global ranking method.
  */
-public abstract class RankDoc extends ScoreDoc implements Writeable {
+public abstract class RankDoc extends ScoreDoc implements NamedWriteable {
```

Comment:
How do you get away with this without adjusting all current serialization for `RankDoc`? 

Switching from writeable to namewriteable might break some things? Maybe I missed it.

Reply:
The only concrete implementations up until now where the `RRFRankDoc` and `TestRankDoc` (used in tests). In both cases I added the impl for the `getWriteableName` method and added this to `RRFRankPlugin`'s exported `getNamedWriteables` . 

The issue was that up until now we explicitly mapped to a `RRFRankDoc` in `RRFRankShardResult`, however now the `fetch` phase could in theory accept any arbitrary concrete implementation of the `RankDoc` class.  So this is why I had to update to a `NamedWriteable`, to make use of (in `RankDocShardInfo`) 
```
        rankDocs = in.readMap(StreamInput::readVInt, v -> v.readNamedWriteable(RankDoc.class));

```
Is there any additional serialization needed that I missed? :/

Reply:
> Is there any additional serialization needed that I missed? :/

I don't know. The retrievers and rank code has evolved outside of my purview for a long time.